packages/geoprocessing/src/types.ts

```diff
@@ -88,7 +89,28 @@ type GeoprocessingServiceType = "javascript" | "container";
 /** Expected public service metadata for each function */
 export interface GeoprocessingServiceMetadata
   extends GeoprocessingHandlerOptions {
-  rateLimit: boolean;
+  restrictedAccess?: boolean; // NOT IMPLEMENTED?
+  /** Seconds */
+  medianDuration: number;
+  /** USD */
+  medianCost: number;
+  endpoint: string;
+  type: GeoprocessingServiceType;
+  // for low-latency clientside processing and offline use
+  // v2 or later
+  clientSideBundle?: ClientCode;
+  // e.g. [sensitive-project.seasketch.org]
+  issAllowList: string[];
+  rateLimited: boolean;
+  rateLimit: number;
+  rateLimitPeriod: RateLimitPeriod;
+  rateLimitConsumed: number;
+  // if set, requests must include a token with an allowed issuer (iss)
+  uri?: string; // NOT IMPLEMENTED?
+}
+
+export interface PreprocessingServiceMetadata
+  extends PreprocessingHandlerOptions {
```

Comment:
This didn't exist before, preprocessing service meta was lumped into geoprocessing service meta.  If they have the same props, then they should get a base interface